src/entities/group/group.controller.spec.ts

```diff
@@ -0,0 +1,267 @@
+import { Request, Response } from 'express';
+
+import { groupController } from './group.controller';
+import { groupService } from './group.service';
+
+import { ErrorCodes } from '../../errors';
+import { GroupModel } from '../../db/models';
+
+describe('Group Controller', () => {
+  let req: Request, res: Response;
+
+  beforeEach(() => {
+    req = {
+      url: 'http://localhost:3000/api/groups',
+    } as Request;
+
+    res = ({
+      send: jest.fn().mockReturnThis(),
+      sendStatus: jest.fn().mockReturnThis(),
+      status: jest.fn().mockReturnThis(),
+    } as unknown) as Response;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('getGroups', () => {
+    it('should call group service with custom limit if it is in query params', async () => {
+      req.url = 'http://localhost:3000/api/users?limit=2';
+
+      jest
+        .spyOn(groupService, 'getAll')
+        .mockImplementation(() => Promise.resolve([]));
+      await groupController.getGroups(req, res);
+
+      expect(groupService.getAll).toHaveBeenCalledWith(2);
```

Comment:
It that correct that groupsService.getAll called twice?

Reply:
it's `toHaveBeenCalledWith`, not `toHaveBeenCalledTimes` :)